### PR TITLE
Adds throughput benchmarking scripts for different learning workflows

### PR DESCRIPTION
**Pull request by kellyguo11**
# Description
- Removes unneeded extensions from headless app file
- Limits CPU threads when running multi-GPU training to minimize thread context switching
- Apply multi GPU renderer setting to avoid unnecessary GPU context initialization
- Decreases 8k SH startup time on 8 GPUs from > 5 min to 1 min
- Adds benchmarking scripts

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/omni/isaac/lab/app/app_launcher.py [hunk 1]
```diff
@@ -464,6 +464,15 @@ def _config_resolution(self, launcher_args: dict):
             if distributed_train:
                 self.device_id = self.local_rank
                 launcher_args["multi_gpu"] = False
+            # limit CPU threads to minimize thread context switching
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/app/app_launcher.py:467]: NIT: Can we have slightly more explanation on why this is good to do? Just for future us to know it as well :)

File: source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py [hunk 2]
```diff
@@ -60,14 +60,23 @@ class Timer(ContextDecorator):
     Reference: https://gist.github.com/sumeet/1123871
     """
 
-    def __init__(self, msg: str | None = None):
+    timing_info = dict()
+    """ Used to log timing instances for retrieval from outside
+        of the Timer class. Times will be logged if a non-empty
+        name is provided on stop.
+    """
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py]: Maybe stating it as class-var explicitly makes it clearer: https://peps.python.org/pep-0526/#class-and-instance-variable-annotations

```suggestion
    timing_info: ClassVar[dict[str, float]] = dict()
    """Dictionary for storing the elapsed time per timer instances globally.
    
    This dictionary logs the timer information. The keys are the names given to the timer class
    at its initialization. If no :attr:`name` is passed to the constructor, no time
    is recorded in the dictionary.
    """
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py [hunk 3]
```diff
@@ -118,6 +127,9 @@ def stop(self):
         self._elapsed_time = self._stop_time - self._start_time
         self._start_time = None
 
+        if self._name:
+            Timer.timing_info[self._name] = self._elapsed_time
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py:133]: Just thinking: Do we also want to store the count -- number of times that timer was called? That makes it simple to compute the average?

File: source/extensions/omni.isaac.lab/test/terrains/check_terrain_importer.py [hunk 4]
```diff
@@ -79,6 +80,8 @@
 from omni.isaac.lab.terrains.terrain_importer import TerrainImporter
 from omni.isaac.lab.utils.assets import ISAAC_NUCLEUS_DIR
 
+omni.kit.app.get_app().get_extension_manager().set_extension_enabled_immediate("omni.kit.primitive.mesh", True)
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/test/terrains/check_terrain_importer.py]: Let's use Isaac Sim's enable_extension function instead?

```python
from omni.isaac.core.utils.extensions import enable_extension

enable_extension("omni.kit.primitive.mesh")
```

File: source/standalone/workflows/benchmarks/utils.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    return log_data
+
+
+#############################
+# logging benchmark metrics #
+#############################
+
+
+def log_min_max_mean_stats(benchmark: BaseIsaacBenchmark, values: dict):
```
Comment (Mayankm96) [inline source/standalone/workflows/benchmarks/utils.py]: NIT: Some docstrings would be nice for all of these though functions are simple enough to read and understand.

**Review by Mayankm96 — COMMENTED** [review]
Mostly looks fine. Would be nice if we can separate the two things into different MRs (startup time and benchmarking) since they are independent of each other (and fairly important changes/additions).

File: source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py (hunk 3 above)
Comment (kellyguo11) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py:133]: hmm I thought it just captures a one-time elapsed time?

Comment (kellyguo11) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py:133]: ah nvm, you mean like storing a list of the times with the same name?

**Review by Dhoeller19 — APPROVED** [review]

File: source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py (hunk 3 above)
Comment (Dhoeller19) [reply source/extensions/omni.isaac.lab/omni/isaac/lab/utils/timer.py:133]: I think it is fine for now as you would make the timer harder to understand.

**Pull request merged**